test/csuite/random_directio/main.c

```diff
@@ -0,0 +1,1127 @@
+		reverse(keyid);
+	testutil_assert(keyid_size + 4 <= buf_size);
+	large_size = buf_size - 4 - keyid_size;
+	testutil_check(__wt_snprintf(buf, buf_size,
+	    "%s" KEY_SEP "%1.1x" KEY_SEP "%.*s",
+	    keyid, threadid, (int)large_size, large));
+}
+
+/*
+ * gen_table_name --
+ *	Generate a table name used for the schema test.
+ */
+static void
+gen_table_name(char *buf, size_t buf_size, uint64_t id, uint32_t threadid)
+{
+	testutil_check(__wt_snprintf(buf, buf_size,
+	    "table:A%" PRIu64 "-%" PRIu32, id, threadid));
+}
+
+/*
+ * gen_table2_name --
+ *	Generate a second table name used for the schema test.
+ */
+static void
+gen_table2_name(char *buf, size_t buf_size, uint64_t id, uint32_t threadid)
+{
+	testutil_check(__wt_snprintf(buf, buf_size,
+	    "table:B%" PRIu64 "-%" PRIu32, id, threadid));
+}
+
+/*
+ * thread_run --
+ *	Run a writer thread.
+ */
+static WT_THREAD_RET thread_run(void *)
+	WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static WT_THREAD_RET
+thread_run(void *arg)
+{
+	WT_CURSOR *cursor, *rev, *sch;
+	WT_RAND_STATE rnd;
+	WT_SESSION *session;
+	WT_THREAD_DATA *td;
+	size_t lsize;
+	uint64_t i;
+	uint32_t kvsize;
+	int ret;
+	char *buf1, *buf2;
+	char large[LARGE_WRITE_SIZE];
+	bool retryable_error;
+
+	__wt_random_init(&rnd);
+	lsize = sizeof(large);
+	memset(large, 0, lsize);
+
+	td = (WT_THREAD_DATA *)arg;
+	large_buf(large, lsize, td->id, true);
+
+	testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
+	testutil_check(session->open_cursor(session, uri_main, NULL, NULL,
+	    &cursor));
+	testutil_check(session->open_cursor(session, uri_rev, NULL, NULL,
+	    &rev));
+
+	/*
+	 * Split the allocated buffer into two parts, one for
+	 * the key, one for the value.
+	 */
+	kvsize = td->datasize / 2;
+	buf1 = td->data;
+	buf2 = &td->data[kvsize];
+
+	/*
+	 * Continuing writing until we're killed.
+	 */
+	printf("Thread %" PRIu32 "\n", td->id);
+	for (i = 0; ; ++i) {
+	again:
+		retryable_error = false;
+		/*
+		if (i > 0 && i % 10000 == 0)
+			printf("Thread %d completed %d entries\n",
+			    (int)td->id, (int)i);
+		*/
+
+		gen_kv(buf1, kvsize, i, td->id, large, true);
+		gen_kv(buf2, kvsize, i, td->id, large, false);
+
+		testutil_check(session->begin_transaction(session, NULL));
+		cursor->set_key(cursor, buf1);
+		/*
+		 * Every 1000th record write a very large value that exceeds the
+		 * log buffer size.  This forces us to use the unbuffered path.
+		 */
+		if (i % 1000 == 0) {
+			cursor->set_value(cursor, large);
+		} else {
+			cursor->set_value(cursor, buf2);
+		}
+		testutil_check(cursor->insert(cursor));
+
+		/*
+		 * The reverse table has no very large records.
+		 */
+		rev->set_key(rev, buf2);
+		rev->set_value(rev, buf1);
+		testutil_check(rev->insert(rev));
+
+		/*
+		 * If we are doing a schema test, generate operations
+		 * for additional tables.  Each table has a 'lifetime'
+		 * of 4 values of the id.
+		 */
+		if (td->schema_test) {
+			if (!retryable_error && SCHEMA_OP(i, 0)) {
```

Comment:
Every conditional within this is `!retryable_error` so that can be put on line 354 as `if (td->schema_test && !retryable_error) {`. Can the `SCHEMA_OP` be modified so that it can something like `switch ((op = SCHEMA_OP(i))` with cases 0-3 instead of sequential conditionals? Or if that is really what we want, please add a comment describing the testing strategy here.

Reply:
@sueloverso, I've reworked the code and commented, hopefully the next round will be clearer.